vxfreeswitch/voice.py

```diff
@@ -133,9 +133,16 @@ def onChannelExecuteComplete(self, ev):
         if self.request_hang_up:
             return self.hangup()
 
-    def onChannelHangup(self, ev):
+    def onChannelHangupComplete(self, ev):
         log.msg("Channel HangUp")
-        self.vumi_transport.deregister_client(self)
+        try:
+            answered_time = int(ev.get('Caller_Channel_Answered_Time'))
+            hangup_time = int(ev.get('Caller_Channel_Hangup_Time'))
+            duration = hangup_time - answered_time
+        except (TypeError, ValueError):
+            log.warning("Unable to get call duration")
```

Comment:
Might be worthwhile logging some information which will allow us to identify which call the failure is related to.